geode-wan/src/main/java/org/apache/geode/cache/wan/internal/WanCopyRegionFunctionService.java

```diff
@@ -43,7 +45,9 @@
   public boolean init(Cache cache) {
     String WAN_COPY_REGION_FUNCTION_EXECUTION_PROCESSOR_THREAD_PREFIX =
         "WAN Copy Region Function Execution Processor";
-    int WAN_COPY_REGION_FUNCTION_MAX_CONCURRENT_THREADS = 10;
+    int WAN_COPY_REGION_FUNCTION_MAX_CONCURRENT_THREADS = SystemProperty
+        .getProductIntegerProperty(
+            SystemPropertyHelper.WAN_COPY_REGION_FUNCTION_MAX_CONCURRENT_THREADS, 10);
```

Comment:
I seem to be missing some tests here. Or maybe there is one already confirming concurrent threads. Is this something that we want to test out? The behavior when changing the max threads setting? Is there any potential for failure if we run with 1 thread vs 10 threads vs 100 threads?

Also, I don't know how I feel if we have an unlimited thread count setting? Maybe we can initially cap the max threads? Or maybe have another (internal) property that would allow us to change the max-thread cap?

Reply:
I have added a test case to make sure that when the maximum limit is reached the new command will wait for a thread to be free.
I am not sure if it is really necessary to add a cap for max threads.

Reply:
@albertogpz Max threads makes sure that we don't overwhelm a system. As this command can be run at any time (there is no notion of a maintenance mode for Geode) we have to assume that there would be a runtime system impact as we increase the number of threads running a region-copy.